### PR TITLE
Don't crash with --pyargs and a filename that looks like a module

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

File: testing/acceptance_test.py [hunk 1]
```diff
@@ -646,6 +646,12 @@ def test_pyargs_only_imported_once(self, testdir):
         # should only configure once
         assert result.outlines.count("configuring") == 1
 
+    def test_pyargs_filename_looks_like_module(self, testdir):
+        testdir.tmpdir.join("conftest.py").ensure()
+        testdir.tmpdir.join("t.py").write("def test(): pass")
+        result = testdir.runpytest("--pyargs", "t.py")
```
Comment (nicoddemus) [inline testing/acceptance_test.py]: Hmm what happens here? Does it run the file? Is that what's expected?

Comment (asottile) [reply testing/acceptance_test.py]: it runs the file, yes -- it's at least consistent with the 4.6.x behaviour -- I didn't really consider whether this is sane / correct or not :laughing: 

I've never really worked with `--pyargs` so I'm not sure what the expected thing is here

Comment (nicoddemus) [reply testing/acceptance_test.py]: Well if it works like 4.6.x then that's fine by me! 

Thanks for the quick fix!

**Comment by nicoddemus** [comment]
> Only missing a CHANGELOG entry.

That is, unless that bug was never released in the first place.

**Comment by asottile** [comment]
> > Only missing a CHANGELOG entry.
> 
> That is, unless that bug was never released in the first place.

:this:

**Comment by asottile** [comment]
@nicoddemus approve? :D

**Comment by nicoddemus** [comment]
Oh sorry!

**Comment by nicoddemus** [comment]
Feel free to cut the 5.0.0 release; I won't have time until later today

**Comment by asottile** [comment]
> Feel free to cut the 5.0.0 release; I won't have time until later today

sounds good! I'll take a stab at it and fix up the things in the original PR :+1:

**Comment by nicoddemus** [comment]
Awesome 🎉

**Pull request merged**